CMakeLists.txt

```diff
@@ -86,11 +90,27 @@ if(NOT asio_POPULATED AND NOT TARGET asio)
 endif()
 
 if(ASIO_ENABLE_EXAMPLES)
+  enable_testing()
+
   add_executable(asio_example_cpp03_timers_time_t_timer ${asio_SOURCE_DIR}/asio/src/examples/cpp03/timers/time_t_timer.cpp)
   target_link_libraries(asio_example_cpp03_timers_time_t_timer PRIVATE asio::asio)
+  add_test(NAME asio_example COMMAND asio_example_cpp03_timers_time_t_timer)
 
-  # todo : add all examples
+  # TODO: add all examples
 
   add_custom_target(asio_examples DEPENDS asio_example_cpp03_timers_time_t_timer)
+endif()
 
+if(ASIO_USE_CPM)
+  packageProject(
+    NAME asio
+    VERSION ${PROJECT_VERSION}
```

Comment:
If you want a different version for your project, you have to define a Version for ASIO here.